src/common/xattr.h

```diff
@@ -19,12 +19,14 @@
 extern "C" {
 #endif
 
+#if defined(__linux__)
```

Comment:
The definition was already there, I just wrapped it in ifdef because might case redefinition issues, and compiler warnings... 
But I think ENOATTR is not used in ANY of the code other than compat.h.
And there it is to create compatibility with FreeBSD.
So yes, I think this whole piece could/should go.??